pkg-manager/resolve-dependencies/src/resolvePeers.ts

```diff
@@ -108,17 +108,32 @@ export function resolvePeers<T extends PartialResolvedPackage> (
     node.children = mapValues((childNodeId) => pathsByNodeId[childNodeId] ?? childNodeId, node.children)
   })
 
-  const dependenciesByProjectId: { [id: string]: { [alias: string]: string } } = {}
+  const dependenciesByProjectId: { [id: string]: Record<string, string> } = {}
   for (const { directNodeIdsByAlias, id } of opts.projects) {
     dependenciesByProjectId[id] = mapValues((nodeId) => pathsByNodeId[nodeId], directNodeIdsByAlias)
   }
   if (opts.dedupePeerDependents) {
-    const depPathsMap = deduplicateDepPaths(depPathsByPkgId, depGraph)
-    Object.values(depGraph).forEach((node) => {
-      node.children = mapValues((childDepPath) => depPathsMap[childDepPath] ?? childDepPath, node.children)
-    })
+    let duplicates = Object.values(depPathsByPkgId).filter((item) => item.length > 1)
+    let maxPasses = duplicates.length
+    let allDepPathsMap: Record<string, string> = {}
+
+    function deduplicationPass () {
+      const res = deduplicateDepPaths(duplicates, depGraph)
+      const { depPathsMap } = res
+      duplicates = res.remainingDuplicates
+
+      Object.values(depGraph).forEach((node) => {
+        node.children = mapValues((childDepPath) => depPathsMap[childDepPath] ?? childDepPath, node.children)
+      })
```

Comment:
what about this one? Can this be moved out from the loop?

Reply:
No, that is required for `pkg-manager/resolve-dependencies/src/resolvePeers.ts:196-197` to get the dependency path for any peers that have been deduplicated in a previous pass.

Reply:
This is a pretty expensive operation. Can at least `Object.values(depGraph)` be moved to a variable?